lib/services/spam-detection.js

```diff
@@ -118,7 +118,8 @@ class SpamDetection {
     return this.isSimilarityAboveThreshold(matchPercents);
   }
 
-  checkListOfRecentUrlsForSpam(message, urlList) {
+  checkListOfRecentUrlsForSpam(message, urlList, mentionUser = null) {
+    if (mentionUser && !this.messageMatching.mentionsUser(message, mentionUser)) return false;
```

Comment:
`mentionsUser()` already accounts for falsy values of `user`, so the first check isn't necessary.